src/components/Modal/__tests__/modal-tests.jest.js

```diff
@@ -0,0 +1,175 @@
+      const component = renderComponent({ openOnStart: true, isAlertDialog: true });
+      fireEvent.keyDown(queryAlertModal(component), { key: "Escape", code: "Escape" });
+      fireEvent.click(await findOverlay(component));
+
+      expect(queryAlertModal(component)).toBeInTheDocument();
+    });
+  });
+
+  describe("accessibility", () => {
+    describe("container", () => {
+      it("should have relevant aria attributes when shown", () => {
+        const component = renderComponent({ openOnStart: true });
+        const modal = queryModal(component);
+        expect(modal).toHaveAttribute("id");
+        expect(modal).toHaveAttribute("aria-modal");
+        expect(modal).toHaveAttribute("aria-labelledby");
+        expect(modal).not.toHaveAttribute("aria-hidden");
+        expect(modal.getAttribute("role")).toMatch("dialog");
+      });
+
+      it("should have relevant aria attributes when hidden", () => {
+        const component = renderComponent();
+        const modal = queryClosedModal(component);
+        expect(modal).toHaveAttribute("id");
+        expect(modal).toHaveAttribute("aria-modal");
+        expect(modal).toHaveAttribute("aria-labelledby");
+        expect(modal).toHaveAttribute("aria-hidden");
+        expect(modal.getAttribute("role")).toEqual("dialog");
+      });
+
+      it("should have relevant aria attributes when in alert mode", () => {
+        const component = renderComponent({ isAlertDialog: true });
+        const modal = queryClosedModal(component);
+        expect(modal.getAttribute("role")).toEqual("alertdialog");
+      });
+    });
+
+    describe("modal", () => {
+      it("should have relevant attributes", function () {
+        const component = renderComponent();
+        const title = component.queryByText(MODAL_TITLE_TEXT);
+        expect(title).toHaveAttribute("id");
+        expect(title.getAttribute("role")).toEqual("heading");
+      });
+    });
+
+    describe("title", () => {
+      it("should have relevant attributes", function () {
+        const component = renderComponent({ openOnStart: true });
+        const title = component.queryByRole("document");
+        expect(title).toBeInTheDocument();
+      });
+    });
+
+    describe("close button", () => {
+      it("should have relevant attributes", function () {
+        const component = renderComponent({ openOnStart: true });
+        const closeButton = queryCloseButton(component);
+        expect(closeButton).toHaveAttribute("aria-label");
+      });
+    });
+  });
+
+  describe("rendering", () => {
+    it("should render a title if the title prop is provided", () => {
+      const TEST_TITLE = "test title";
+      const component = renderComponent({ openOnStart: true, title: TEST_TITLE });
+      const title = component.queryByText(TEST_TITLE);
+      expect(title).toBeInTheDocument();
+    });
+
+    it("should render ModalHeader if passed as children", () => {
+      const TEST_TITLE = "test title";
```

Comment:
I'm not sure - but It's looks like duplication for Snapshot tests ...

Reply:
+1